File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/card/ui/PaymentCardPresenterModule.java
package com.fintechplatform.ui.card.ui;

import com.fintechplatform.api.card.api.PaymentCardAPI;
import com.fintechplatform.ui.card.db.PaymentCardPersistenceDB;
import com.fintechplatform.ui.models.DataAccount;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class PaymentCardPresenterModule {
    private PaymentCardContract.View view;
    private DataAccount dataAccount;

    public PaymentCardPresenterModule(PaymentCardContract.View view, DataAccount dataAccount) {
        this.view = view;
        this.dataAccount = dataAccount;
    }

    @Provides
    @Singleton
    PaymentCardContract.Presenter providesPaymentCardPresenter(PaymentCardAPI api, PaymentCardPersistenceDB paymentCardPersistenceDB) {
        return new PaymentCardPresenter(view, api, dataAccount, paymentCardPersistenceDB);
    }

}

File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/card/di/PaymentCardPresenterModule.kt
package com.fintechplatform.ui.card.di

import com.fintechplatform.api.card.api.PaymentCardAPI
import com.fintechplatform.ui.card.PaymentCardContract
import com.fintechplatform.ui.card.PaymentCardPresenter
import com.fintechplatform.ui.card.db.PaymentCardPersistenceDB
import com.fintechplatform.ui.models.DataAccount
import dagger.Module
import dagger.Provides
import javax.inject.Singleton


@Module
class PaymentCardPresenterModule(private val view: PaymentCardContract.View, private val dataAccount: DataAccount) {

    @Provides
    @Singleton
    internal fun providesPaymentCardPresenter(api: PaymentCardAPI, paymentCardPersistenceDB: PaymentCardPersistenceDB): PaymentCardContract.Presenter =
        PaymentCardPresenter(view, api, dataAccount, paymentCardPersistenceDB)

}